Fix doc refs to StratifiedKFold and KFold

proinsias opened the pull request:
There are several cases throughout the documentation where
`StratifiedKFold` and `KFold` are referred to outside the
`cross_validation` module, and without the proper package prefix. This
breaks the automatic links to the appropriate doc page.

Also fixed "`StratifiedKFold` used" -> "`StratifiedKFold` is used".

glouppe commented:
Thanks! however `sklearn.cross_validation` has been marked as deprecated. We should now refer to `sklearn.model_selection` instead.

agramfort commented on sklearn/calibration.py:
```diff
@@ -63,8 +63,9 @@ class CalibratedClassifierCV(BaseEstimator, ClassifierMixin):
         For integer/None inputs, if ``y`` is binary or multiclass,
-        :class:`StratifiedKFold` used. If ``y`` is neither binary nor
-        multiclass, :class:`KFold` is used.
+        :class:`sklearn.cross_validation.StratifiedKFold` is used. If ``y`` 
```
revert. it was good.

also cross_validation module is deprecated

proinsias committed:
Fix doc refs to StratifiedKFold and KFold

There are several cases throughout the documentation where
`StratifiedKFold` and `KFold` are referred to outside the
`model_selection` module, and without the proper package prefix. This
breaks the automatic links to the appropriate doc page.

Also fixed "`StratifiedKFold` used" -> "`StratifiedKFold` is used".

proinsias commented:
Updated to refer to `sklearn.model_selection` instead.

Re @agramfort's comment re `calibration.py` – when I look at the online doc, I don't see the appropriate link. What am I missing?

agramfort commented:
sorry I don't get the quesion. What link?

proinsias commented:
@agramfort – Check out [cross_val_predict](http://scikit-learn.org/stable/modules/generated/sklearn.cross_validation.cross_val_predict.html#sklearn.cross_validation.cross_val_predict) which as the correct link for [StratifiedKFold](http://scikit-learn.org/stable/modules/generated/sklearn.cross_validation.StratifiedKFold.html#sklearn.cross_validation.StratifiedKFold). I edited `CalibratedClassifierCV` so that this link will be generated.

agramfort commented on sklearn/calibration.py:
```diff
@@ -63,8 +63,9 @@ class CalibratedClassifierCV(BaseEstimator, ClassifierMixin):
         For integer/None inputs, if ``y`` is binary or multiclass,
-        :class:`StratifiedKFold` used. If ``y`` is neither binary nor
-        multiclass, :class:`KFold` is used.
+        :class:`sklearn.model_selection.StratifiedKFold` is used. If ``y`` 
```
:class:`StratifiedKFold`

should be enough. At least in cross_val_predict it's enough. I don't think we should expand
full import.

proinsias replied:
It's not enough to generate the link in the current version of the docs for [CalibratedClassifierCV](http://scikit-learn.org/stable/modules/generated/sklearn.calibration.CalibratedClassifierCV.html). 

It works in `cross_val_predict`, but that class is in the same module as `StratifiedKFold`. 

Full import is used elsewhere in the documentation, e.g., in `/preprocessing/data.py`:
(e.g. as part of a preprocessing :class:`sklearn.pipeline.Pipeline`)
to give the correct link, as seen [here](http://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.scale.html).

agramfort commented:
got it. Thx @proinsias
